ExamRuns: populate run for pilot period

rhysyngsun opened the pull request:
#### What are the relevant tickets?
Part of #3085

#### What's this PR do?
This PR populates data for the pilot period into `ExamRun` and updates our exam authorization data writers to use this new structure.

#### How should this be manually tested?
- Create an `ExamAuthorization` for your user
- Run the migrations, verify your existing authorization got assigned to the pilot exam run.
- Create an exam run and a new authorization assigned for that run:
```python
from exams.factories import *
from exams.models import *
exam_run = ExamRunFactory.create(user=user)
ExamAuthorization.create(exam_run=exam_run, user=user, status=ExamAuthorization.STATUS_SUCCESS)
```
- Run this command:
```python
from exams import tasks
exams.export_exam_authorizations()
```
- Check the output and ensure it output a row for your record and the there is no `Modules` column.

#### What GIF best describes this PR or how it makes you feel?
![](https://media.giphy.com/media/13sozYO4hmSMUw/giphy.gif)

noisecapella commented on exams/migrations/0008_populate_exam_run.py:
```diff
@@ -0,0 +1,42 @@
+                date_first_eligible=PILOT_ELIGIBILITY_START_DATE,
+                date_last_eligible=PILOT_ELIGIBILITY_END_DATE,
+            )
+        exam_auth.exam_run = runs_by_course_id[exam_auth.course_id]
```
Where are we populating `runs_by_course_id`?

rhysyngsun replied:
Fixed.

noisecapella commented on exams/migrations/0008_populate_exam_run.py:
```diff
@@ -0,0 +1,42 @@
+
+from django.db import migrations
+
+PILOT_SERIES_CODE = 'PILOT'
```
Is this setting up `ExamRun`s for a specific course?

rhysyngsun replied:
This is setting them up for the pilot, which was over two courses. The value is arbitrary because we'll never use it for those records since the exams already happened.

rhysyngsun commented:
@noisecapella addressed your feedback, additionally fixed the issue where in the admin console `exam_series_code` was readonly when creating a new `ExamRun`.

noisecapella commented on exams/migrations/0008_populate_exam_run.py:
```diff
@@ -0,0 +1,42 @@
+                date_last_eligible=PILOT_ELIGIBILITY_END_DATE,
+            )
+        exam_auth.exam_run = runs_by_course_id[exam_auth.course_id]
+        exam_auth.save(updated_fields=('exam_run',))
```
I got an exception on this line:

    Operations to perform:
      Apply all migrations: admin, auth, authtoken, cms, contenttypes, courses, dashboard, ecommerce, exams, financialaid, grades, mail, profiles, roles, search, sessions, social_django, taggit, wagtailadmin, wagtailcore, wagtaildocs, wagtailembeds, wagtailforms, wagtailimages, wagtailredirects, wagtailsearch, wagtailusers
    Running migrations:
      Applying exams.0008_populate_exam_run.../usr/local/lib/python3.5/site-packages/django/db/models/fields/__init__.py:1430: RuntimeWarning: DateTimeField ExamRun.date_first_schedulable received a naive datetime (2017-03-06 00:00:00) while time zone support is active.
      RuntimeWarning)
    /usr/local/lib/python3.5/site-packages/django/db/models/fields/__init__.py:1430: RuntimeWarning: DateTimeField ExamRun.date_last_schedulable received a naive datetime (2017-03-30 00:00:00) while time zone support is active.
      RuntimeWarning)
    Traceback (most recent call last):
      File "./manage.py", line 14, in <module>
        execute_from_command_line(sys.argv)
      File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
        utility.execute()
      File "/usr/local/lib/python3.5/site-packages/django/core/management/__init__.py", line 359, in execute
        self.fetch_command(subcommand).run_from_argv(self.argv)
      File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 294, in run_from_argv
        self.execute(*args, **cmd_options)
      File "/usr/local/lib/python3.5/site-packages/django/core/management/base.py", line 345, in execute
        output = self.handle(*args, **options)
      File "/usr/local/lib/python3.5/site-packages/django/core/management/commands/migrate.py", line 204, in handle
        fake_initial=fake_initial,
      File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 115, in migrate
        state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
      File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 145, in _migrate_all_forwards
        state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
      File "/usr/local/lib/python3.5/site-packages/django/db/migrations/executor.py", line 244, in apply_migration
        state = migration.apply(state, schema_editor)
      File "/usr/local/lib/python3.5/site-packages/django/db/migrations/migration.py", line 129, in apply
        operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
      File "/usr/local/lib/python3.5/site-packages/django/db/migrations/operations/special.py", line 189, in database_forwards
        self.code(from_state.apps, schema_editor)
      File "/src/exams/migrations/0008_populate_exam_run.py", line 32, in populate_pilot_runs
        exam_auth.save(updated_fields=('exam_run',))
    TypeError: save() got an unexpected keyword argument 'updated_fields'

noisecapella commented:
I'm getting some warnings here regarding time zones: 

> /usr/local/lib/python3.5/site-packages/django/db/models/fields/__init__.py:1430: RuntimeWarning: DateTimeField ExamRun.date_last_schedulable received a naive datetime (2017-03-30 00:00:00) while time zone support is active.
  RuntimeWarning)

I don't think it's a big deal though since these are fake dates anyway
